integration/test-integration/src/test/java/org/wso2/choreo/connect/tests/setup/withapim/ApimRestartExecutor.java

```diff
@@ -24,32 +24,32 @@
 import org.testng.annotations.Test;
 import org.wso2.choreo.connect.tests.context.ApimInstance;
 import org.wso2.choreo.connect.tests.context.CCTestException;
+import org.wso2.choreo.connect.tests.context.CcInstance;
 import org.wso2.choreo.connect.tests.util.TestConstant;
 import org.wso2.choreo.connect.tests.util.Utils;
 
+import java.io.IOException;
 import java.util.concurrent.Callable;
 import java.util.concurrent.TimeUnit;
 
 public class ApimRestartExecutor {
-    private static final Logger log = LoggerFactory.getLogger(ApimRestartExecutor.class);
+    CcInstance ccInstance;
```

Comment:
If we restart Choreo Connect also, what we test would be the Choreo Connect startup pull flow. Therefore, implementation for the retry mechanism will not be tested.

Reply:
The choreo connect starts up way before apim does. Then it keeps retrying until apim is functional.